src/formatter/json_formatter.js

```diff
@@ -156,7 +156,10 @@ export default class JsonFormatter extends Formatter {
         data.result.duration = testStep.result.duration
       }
       if (status === Status.FAILED && exception) {
-        data.result.error_message = exception.stack || exception
+        let message = exception.message || ''
+        let stack = exception.stack || ''
+        message = _.includes(stack, message) ? '' : message + '\n'
+        data.result.error_message = message + stack || exception
```

Comment:
for consistency over the type of error_message, I would suggest to have a `JSON.stringify(exception)`. It would be a rare exception, but at least we know error_message will be a string all the time.

Reply:
I omitted the JSON.stringify in the pull request because that would change the spec (JSON.stringify add escaped quotes to string object):

https://github.com/cucumber/cucumber-js/blob/267725e3a8b00e28336ef75727d7487e5a6739ab/src/formatter/json_formatter_spec.js#L128

https://github.com/cucumber/cucumber-js/blob/267725e3a8b00e28336ef75727d7487e5a6739ab/src/formatter/json_formatter_spec.js#L141

Adding a type check would enforce a string without changing the spec.
I wouldn't add more code if not really needed

```
data.result.error_message = message + stack || _.isString(exception) ? exception :  JSON.stringify(exception)
```

Reply:
@finaruioana 

This may be a better solution, what do you think?


```javascript
if(_.isString(exception)) {
  data.result.error_message = exception
} else {
  let {message = '', stack = ''} = exception
  message = _.includes(stack, message) ? '' : message + '\n'
  data.result.error_message = message + stack || JSON.stringify(exception)
}
```